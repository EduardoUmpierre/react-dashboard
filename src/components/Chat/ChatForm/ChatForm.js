import React from 'react'
import './ChatForm.scss'

const chatForm = ({ inputValue, onInputChange, onFormSubmit, isDisabled }) => (
    <form className='chat-form' onSubmit={e => onFormSubmit(e)}>
        <input
            type='text'
            placeholder='Type your message here...'
            value={inputValue}
            onChange={e => onInputChange(e)}
            required
        />
        <button type='submit' disabled={isDisabled}>
            Send
        </button>
    </form>
)

export default chatForm
