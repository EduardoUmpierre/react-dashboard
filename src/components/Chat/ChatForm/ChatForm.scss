.chat-form {
    border-top: 1px solid #eee;
    display: flex;
    padding: 0.625rem;

    > input {
        border: 1px solid #eee;
        border-radius: 3px 0 0 3px;
        color: #777783;
        padding: 0.5rem 0.625rem;
        width: 100%;
    }

    > button {
        border: none;
        border-radius: 0 3px 3px 0;
        color: #fff;
        font-weight: 700;
        padding: 0.5rem 0.875rem;

        &:not([disabled]) {
            background-color: #8ad919;
            cursor: pointer;
        }
    }

    > input,
    > button {
        font-size: 0.875rem;
    }
}
