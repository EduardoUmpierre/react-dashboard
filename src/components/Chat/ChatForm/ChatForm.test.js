import React from 'react'
import { shallow } from 'enzyme'
import ChatForm from './ChatForm'

describe('<ChatForm />', () => {
    it('renders correctly', () => {
        const component = shallow(<ChatForm inputValue='' isDisabled={false} />)
        expect(component).toMatchSnapshot()
    })

    it('button is disabled', () => {
        const component = shallow(<ChatForm inputValue='' isDisabled={true} />)
        expect(component.find('button').props().disabled).toBe(true)
    })

    it('button is enabled', () => {
        const component = shallow(<ChatForm inputValue='' isDisabled={false} />)
        expect(component.find('button').props().disabled).toBe(false)
    })
})
