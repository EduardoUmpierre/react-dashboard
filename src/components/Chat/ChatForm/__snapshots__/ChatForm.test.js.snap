// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatForm /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <chatForm
    inputValue=""
    isDisabled={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          placeholder="Type your message here..."
          required={true}
          type="text"
          value=""
        />,
        <button
          disabled={false}
          type="submit"
        >
          Send
        </button>,
      ],
      "className": "chat-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "Type your message here...",
          "required": true,
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Send",
          "disabled": false,
          "type": "submit",
        },
        "ref": null,
        "rendered": "Send",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            placeholder="Type your message here..."
            required={true}
            type="text"
            value=""
          />,
          <button
            disabled={false}
            type="submit"
          >
            Send
          </button>,
        ],
        "className": "chat-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Type your message here...",
            "required": true,
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Send",
            "disabled": false,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Send",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
